github/ci/testgrid/hack/check.sh

```diff
@@ -3,16 +3,42 @@
 set -euo pipefail
 
 BASEDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
-source ${BASEDIR}/common.sh
+PROJECT_INFRA_ROOT=$(readlink -f --canonicalize ${BASEDIR}/../../../..)
+if [ -d ${PROJECT_INFRA_ROOT}/../../kubernetes/test-infra ]; then
+    TEST_INFRA_ROOT=$(readlink -f --canonicalize ${PROJECT_INFRA_ROOT}/../../kubernetes/test-infra)
+fi
+
+generate_config(){
+    local prow_config=${PROJECT_INFRA_ROOT}/github/ci/prow-deploy/kustom/base/configs/current/config/config.yaml
+    local job_config=${PROJECT_INFRA_ROOT}/github/ci/prow-deploy/files/jobs
+    local testgrid_dir=${TEST_INFRA_ROOT}/config/testgrids
+    local testgrid_subdir=kubevirt
+    local testgrid_gen_config=$(readlink -f --canonicalize ${BASEDIR}/../gen-config.yaml)
+    local testgrid_default=$(readlink -f --canonicalize ${BASEDIR}/../default.yaml)
+
+    mkdir -p "${testgrid_dir}/${testgrid_subdir}"
+
+    /configurator \
+        --prow-config "${prow_config}" \
+        --prow-job-config "${job_config}" \
+        --output-yaml \
+        --yaml "${testgrid_gen_config}" \
+        --default "${testgrid_default}" \
+        --oneshot \
+        --output "${testgrid_dir}/${testgrid_subdir}/gen-config.yaml"
```

Comment:
Maybe add `--verify-config-file` here as described in the [README](https://github.com/kubernetes/test-infra/blob/master/testgrid/cmd/configurator/README.md#basic-usage)? Or would that break the existing workflow?

Reply:
It would not upload the file, according to the command help:
```
-validate-config-file
    	validate that the given config files are syntactically correct and exit (proto is not written anywhere)
```
In any case, we already have a job that checks the configuration before merging https://github.com/kubevirt/project-infra/blob/master/github/ci/prow-deploy/files/jobs/kubevirt/project-infra/project-infra-presubmits.yaml#L500 at this point all should be already verified.